### PR TITLE
Build a set of configuration files that meet criteria. 

### PR DESCRIPTION
**Pull request by martinstreicher**
Do not deploy to S3 (yet). Run tests as the script. Do not merge this yet.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -55,6 +56,7 @@
     "karma-spec-reporter": "0.0.20",
     "karma-webpack": "^1.7.0",
     "lodash": "^4.0.0",
+    "nodemon": "1.8.1",
```
Comment (colinrymer) [inline package.json]: What is nodemon being used for?

**Comment by martinstreicher** [comment]
@Teapane @c0 @colinrymer 

So far:
- `npm run lint` fails. 

- I don't know if `semantic-release` works. 

- Need to change the Travis config to deploy to S3 only after tests pass and arguably only from a particular branch. 

- Also need a config to do the right thing using `semantic-release`

File: package.json (hunk 1 above)
Comment (martinstreicher) [inline package.json]: I am not using it per se. It came up as an dependency in the `npm install` step.

**Comment by Teapane** [comment]
@martinstreicher pushed a change for linting.

**Comment by Teapane** [comment]
@martinstreicher I pushed up a few changes. The only thing I am not :100:% on is the global `ENV` vars. Maybe Colin could weigh in? Otherwise, I am :+1:

**Comment by martinstreicher** [comment]
@colinrymer @Teapane This is ready for a review. This should get us to the point where a semantic release is sent to npm. It does not yet deploy to S3. 

Some questions:

1. The latest release was 1.2.2. Can we make the first release to npm 1.3.0 since the current code has a new feature in it. How do we set the baseline version number for semantic-release to use? 

2. Once a release to npm occurs, I can likely find the version number of that release and use those numbers to perform the release to S3. Previously, the version number came from package.json as it was a literal string. With semantic-release, there is a placeholder and the ultimate version number is left in npm. I'll take those numbers and a modified deploy.js script and push to S3. OK?

**Comment by martinstreicher** [comment]
To make things a little easier to review, I am going to close this PR and make two new ones. One PR will be for lint and Travis changes; the other PR will be the feature change.

**Pull request closed without merging**